draft-ietf-quic-transport.md

```diff
@@ -208,1349 +223,1114 @@ x (*) ...
-Destination Connection ID of subsequent packets that it sends.
+An endpoint is expected to send another STOP_SENDING frame if a packet
+containing a previous STOP_SENDING is lost.  However, once either all stream
+data or a RST_STREAM frame has been received for the stream - that is, the
+stream is in any state other than "Recv" or "Size Known" - sending a
+STOP_SENDING frame is unnecessary.
 
-A server MAY send Retry packets in response to Initial and 0-RTT packets.  A
-server can either discard or buffer 0-RTT packets that it receives.  A server
-can send multiple Retry packets as it receives Initial or 0-RTT packets.
 
-A client MUST accept and process at most one Retry packet for each connection
-attempt.  After the client has received and processed an Initial or Retry packet
-from the server, it MUST discard any subsequent Retry packets that it receives.
+# Flow Control {#flow-control}
 
-Clients MUST discard Retry packets that contain an Original Destination
-Connection ID field that does not match the Destination Connection ID from its
-Initial packet.  This prevents an off-path attacker from injecting a Retry
-packet.
+It is necessary to limit the amount of data that a sender may have outstanding
+at any time, so as to prevent a fast sender from overwhelming a slow receiver,
+or to prevent a malicious sender from consuming significant resources at a
+receiver.  To this end, QUIC employs a credit-based flow-control scheme similar
+to that in HTTP/2 {{?HTTP2}}.  A receiver advertises the number of octets it is
+prepared to receive on a given stream and for the entire connection.  This leads
+to two levels of flow control in QUIC:
 
-The client responds to a Retry packet with an Initial packet that includes the
-provided Retry Token to continue connection establishment.
+* Stream flow control, which prevents a single stream from consuming the entire
+  receive buffer for a connection.
 
-A client sets the Destination Connection ID field of this Initial packet to the
-value from the Source Connection ID in the Retry packet. Changing Destination
-Connection ID also results in a change to the keys used to protect the Initial
-packet. It also sets the Token field to the token provided in the Retry. The
-client MUST NOT change the Source Connection ID because the server could include
-the connection ID as part of its token validation logic (see {{tokens}}).
+* Connection flow control, which prevents senders from exceeding a receiver's
+  buffer capacity for the connection, and
 
-All subsequent Initial packets from the client MUST use the connection ID and
-token values from the Retry packet.  Aside from this, the Initial packet sent
-by the client is subject to the same restrictions as the first Initial packet.
-A client can either reuse the cryptographic handshake message or construct a
-new one at its discretion.
+A data receiver sends MAX_STREAM_DATA or MAX_DATA frames to the sender to
```

Comment:
I think we should add a paragraph here that goes like this:

"Endpoints SHOULD communicate initial values for stream and connection flow control using transport parameters {#transport-params}. Data senders MUST use a value of zero for limits not advertised in the peer's transport parameters."

Reply:
The paragraph above looks good enough to me.  I don't think we need to set defaults (if this were the only thing that is read, a reader would (mostly correctly) infer that the values are always set.